sos/report/plugins/aap_containerized.py

```diff
@@ -0,0 +1,123 @@
+            user_home_directory = os.path.expanduser(f"~{username}")
+            aap_directory_name = self.path_join(user_home_directory, "aap")
+        else:
+            aap_directory_name = self.get_option("directory")
+
+        # Don't collect cert and key files from the installation directory
+        if self.path_exists(aap_directory_name):
+            forbidden_paths = [
+                self.path_join(aap_directory_name, path)
+                for path in [
+                    "containers",
+                    "tls",
+                    "controller/etc/*.cert",
+                    "controller/etc/*.key",
+                    "eda/etc/*.cert",
+                    "eda/etc/*.key",
+                    "gateway/etc/*.cert",
+                    "gateway/etc/*.key",
+                    "hub/etc/*.cert",
+                    "hub/etc/*.key",
+                    "hub/etc/keys/*.pem",
+                    "postgresql/*.crt",
+                    "postgresql/*.key",
+                    "receptor/etc/*.crt",
+                    "receptor/etc/*.key",
+                    "receptor/etc/*.pem",
+                    "redis/*.crt",
+                    "redis/*.key",
+                ]
+            ]
+            self.add_forbidden_path(forbidden_paths)
+            self.add_copy_spec(aap_directory_name)
+        else:
+            self._log_error(f"Directory {aap_directory_name} does not exist")
+
+        # Gather output of following podman commands as user
+        podman_commands = [
+            (f"su - {username} -c 'podman info --debug'", "podman_info"),
+            (f"su - {username} -c 'podman ps -a --format json'",
+                "podman_ps_all_json"),
+        ]
+
+        for command, filename in podman_commands:
+            self.add_cmd_output(command, suggest_filename=filename)
+
+        # Collect AAP container names
+        aap_containers = self._get_aap_container_names(username)
+
+        # Copy podman container log files in plugin sub directory
+        # under aap_containers_log
+        for container in aap_containers:
+            log_file = f"{container}.log"
+            self.add_cmd_output(
+                    f"su - {username} -c 'podman logs {container}'",
+                    suggest_filename=f"{log_file}",
+                    subdir="aap_containers_log"
+            )
+
+    def _get_aap_container_names(self, username):
+        try:
+            cmd = f"su - {username} -c 'podman ps -a --format {{{{.Names}}}}'"
+            cmd_out = self.exec_cmd(cmd)
+            return cmd_out['output'].strip().split("\n")
+        except Exception:
+            self._log_error("Error retrieving Podman containers")
+            return []
+
+    # Check and enable plugin on a AAP Containerized setup
+    def check_enabled(self):
+        ps = self.exec_cmd("ps aux --noheaders")
+        if "awx-manage" in ps["output"] and "aap-gateway" in ps["output"]:
+            return True
+        return False
```

Comment:
Can't this be simple:

           return "awx-manage" in ps["output"] and "aap-gateway" in ps["output"]

?

Can't this provide false positive results when there would be a running process like, say, "grep awx-manage some.log"? Isn't `ps --noheaders axco command` and matching whole line a better check here, or is that already over-complicated..?

But I am OK with the current code.

Reply:
@pmoravec Thank you for the suggestion. Changes amended as requested.